crates/uv-cli/src/lib.rs

```diff
@@ -1559,6 +1571,21 @@ pub struct PipUninstallArgs {
     #[arg(long, value_enum, env = "UV_KEYRING_PROVIDER")]
     pub keyring_provider: Option<KeyringProviderType>,
 
+    /// A list of trusted hosts for SSL connections.
+    ///
+    /// Expects to receive either a hostname (e.g., `localhost`) or a host-port pair
+    /// (e.g., `localhost:8080`).
+    ///
+    /// WARNING: Hosts included in this list will not be verified against the system's certificate
```

Comment:
Would it be reasonable to also warn the user once on the usage of this option / implications? (e.g. mostly makes sense on trusted private registries/mirrors or self-signed corporate pypi proxies)?

Reply:
I think that makes sense. Although since it's opt-in maybe they would already be aware?

Reply:
As in, show a warning on the CLI?

Reply:
I think that's what Sammy means. Iirc, pip doesn't show a warning.

Reply:
Indeed, I was thinking CLI warning.

Reply:
`pip` doesnt show a warning neither should `uv`, it's an opt-in option. It's the same as doing `curl -k https://fakehost.com`, which doesnt have a warning.

uv should show an error if the user tries to install from a non-trusted source without using "--trusted-hosts"

Reply:
Here's an initial stab at it

```
Only use `--trusted-host` in a secure network with verified sources, as it bypasses SSL verification and could expose you to MITM attacks."
```

Reply:
I think gaby is right here. Follow pip's approach.

Reply:
You often have cases like this https://github.com/astral-sh/uv/issues/1339#issuecomment-2232619423 where using the appropriate `UV_NATIVE_TLS` or `SSL_CERT_FILE` could've sufficed instead of a more dangerous option like `--trusted-host`.  This SO's warning speaks for itself https://stackoverflow.com/a/29751768 as to the dangers of blindly using this option over the years.

Reply:
(lurker here) I would prefer the documentation for this command line option to have the warning but the not on actual usage. Rationale: the command is called "trusted-host" as in I trust this thing whatever it may be. This option is also likely to live in the uv.toml or pyproject.toml or env variables. Meaning every single call to uv will have a warning

Reply:
> I trust this thing whatever it may be

Trust is vaguely defined in the name of the option, for example `--trusted-host pypi.org --trusted-host pypi.python.org` hosts may be trusted since they're official, but in an insecure network location it can expose your system to the dangers aformentioned.

Reply:
@charliermarsh

Give the above feedback if we want to have a self-documenting flag and skip the warning, I propose renaming the option to more aligned with other tools e.g. curl's `--insecure`, k8s `--insecure-skip-tls-verify`, docker's `--insecure-registry` etc .?

Hence instead of `--trusted-host`, using something like `--insecure-host` or `--insecure-skip-verify-host`?

uv has a chance to improve this area rather than to stick with the same name which has been easily misunderstood on the security implications over the years. This also can be documented in pip compatibility guide like the other cases.

Reply:
@gaby Yes, I proposed this knowing its the original name used by `pip`. `uv` doesn't have to keep the same name.

Reply:
I think we could say something like `--unsafe-insecure-host`?  I think regardless, we probably need to provide an alias to `--trusted-host`. Though we really feel strongly about a new name we could use our `CompatArgs` utility to suggest using the new flag instead.

Reply:
`--allow-insecure-host`?

Reply:
Just felt like we should reuse our `unsafe` concept from index strategies, but it does get pretty verbose.. `--allow-insecure-host` is okay.

Reply:
I like `--allow-insecure-host` 💯

Reply:
I believe you might be missing the mark with the insecure thing. The connection is secure as in it will still use TLS (transport layer security). It is encrypted by a valid certificate of unknown origin. Note even if you say to trust it, the certificate is still valid.

You have all the benefits of the https connection. Saying this is insecure is comparing this to https which is indeed insecure.

Trusted is the correct word here.

Also this is not only for self signed certificates. There are older systems not recognizing newer CAs such as Let's encrypt (our case for needing this feature).

Reply:
@inoa-jboliveira It's insecure by definition to not have tls verification of the name. Anyone can claim they have a valid certificate otherwise and give you a false sense of security.

Reply:
@samypr100
By this definition, you must show same warning on any HTTP host being used. But it won't happen. 

Just FYI: Regardless of the name you pick by whatever reason you do, I thank the astral team for providing the feature I need the most for wide deploy of uv in my org. You guys rock

Reply:
>By this definition, you must show same warning on any HTTP host being used. But it won't happen.

This option only applies to https connections, not http.